packages/zoe/src/zoeService/types.js

```diff
@@ -49,7 +49,7 @@
  * code breaks.
  * @property {GetConfiguration} getConfiguration
  * @property {GetBundleIDFromInstallation} getBundleIDFromInstallation
- * @property {GetProposalShapeForInvitation} getProposalShapeForInvitation
+ * @property {(invitationHandle: InvitationHandle) => Pattern | undefined} getProposalShapeForInvitation
```

Comment:
This was to get around an import depth problem. `GetProposalShapeForInvitation` is defined in internal-types and here we're describing `ZoeService` which is external. So the type couldn't be resolved. IMO we shouldn't complicate things with callbacks defs like that but I made the minimum necessary change

Reply:
since ZoeService is external, the type should be externally visible. 

> we shouldn't complicate things with callbacks defs like that

say more, please?  Is the argument that it's better/clearer to type out all the details even when a set of parameters and return values are common across multiple calls?  Is this intended as a broad prescription?

Reply:
My argument, which would apply broadly, is that the shared parameter list and return value is better to repeat so that we don't unnecessarily couple the different definitions. TypeScript types structurally, so when two things share the same structure they are as equal as when they share a typedef.

There is value in the developer mental model to things being the same, but in that case we should include the function name as well and say "this method on this object is the same as it is over there. That is best accomplished by a union with a picked subset of the other object.

e.g.,
```
type SomeService = {
  help: () => string;
  doSomething: (str: string) => void;
}
type SomeServiceInternal = Pick<SomeService, 'doSomething'> & {
  doAdminThing: () => AdminFacet;
}
```